---
title: "Site specific TajimaD"
output: html_document
---
```{r}
library(tidyverse)
```

```{r}
theme_set(theme_classic())
df_genotype_tidy <- readRDS("C:/Users/Aysevil/Dropbox/Aarhus Lectures/Project/Data/thymus_genotypes.tidy.rds")
df_sample_info   <- readRDS("C:/Users/Aysevil/Dropbox/Aarhus Lectures/Project/Data/thymus_sample_info.rds")
df_snp_info      <- readRDS("C:/Users/Aysevil/Dropbox/Aarhus Lectures/Project/Data/thymus_snps_info.rds")
```

```{r}
bad_samples <- c("tha151", "tha063","tha238", "tha193") #from replicability
df_genotype_tidy_without_bad_replicants <- df_genotype_tidy %>% 
  filter(!(sample_id %in% bad_samples )) %>% 
  filter(!grepl('b', sample_id))
```


```{r}
sites = c(0)
for (i in 1:length(unique(df_sample_info$site_id))) {
  sites[i] = list((df_sample_info %>% filter(site_id == unique(df_sample_info$site_id)[i]))$sample_id)
}
```

# Create separate dfs per site
```{r}
for (i in 1:length(sites)) {
  df <- df_genotype_tidy_without_bad_replicants %>% 
  filter(sample_id %in% sites[[i]]) %>%  
  group_by(snp) %>% 
  summarise(
            good = sum(!is.na(genotype)),
            rate = good / n(),            
            nb2 = sum(ifelse(is.na(genotype),0, genotype =="2")),
            nb1 = sum(ifelse(is.na(genotype),0, genotype =="1")),
            nb0 = sum(ifelse(is.na(genotype),0, genotype =="0")),
            ) %>%
  left_join(df_snp_info)
  assign(paste("site", i, sep = ""), df)
}
```

```{r}
site1 <- site1 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site2 <- site2 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site3 <- site3 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site4 <- site4 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site5 <- site5 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site6 <- site6 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site7 <- site7 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site8 <- site8 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site9 <- site9 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site10 <- site10 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site11 <- site11 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site12 <- site12 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site13 <- site13 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site14 <- site14 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site15 <- site15 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site16 <- site16 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site17 <- site17 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site18 <- site18 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site19 <- site19 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site20 <- site20 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

site21 <- site21 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

# 70 km away
site22 <- site22 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))

# t.embergeri
site23 <- site23 %>%
  mutate(freq_ref=(2*nb0 + nb1)/(2*good),
         freq_alt=(2*nb2 + nb1)/(2*good))
```

```{r}
site1 <- site1 %>% filter(rate > 0.5)
site2 <- site2 %>% filter(rate > 0.5)
site3 <- site3 %>% filter(rate > 0.5)
site4 <- site4 %>% filter(rate > 0.5)
site5 <- site5 %>% filter(rate > 0.5)
site6 <- site6 %>% filter(rate > 0.5)
site7 <- site7 %>% filter(rate > 0.5)
site8 <- site8 %>% filter(rate > 0.5)
site9 <- site9 %>% filter(rate > 0.5)
site10 <- site10 %>% filter(rate > 0.5)
site11 <- site11 %>% filter(rate > 0.5)
site12 <- site12 %>% filter(rate > 0.5)
site13 <- site13 %>% filter(rate > 0.5)
site14 <- site14 %>% filter(rate > 0.5)
site15 <- site15 %>% filter(rate > 0.5)
site16 <- site16 %>% filter(rate > 0.5)
site17 <- site17 %>% filter(rate > 0.5)
site18 <- site18 %>% filter(rate > 0.5)
site19 <- site19 %>% filter(rate > 0.5)
site20 <- site20 %>% filter(rate > 0.5)
site21 <- site21 %>% filter(rate > 0.5)
site22 <- site22 %>% filter(rate > 0.5)

```

```{r}
site1 <- site1 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site2 <- site2 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site3 <- site3 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site4 <- site4 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site5 <- site5 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site6 <- site6 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site7 <- site7 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site8 <- site8 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site9 <- site9 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site10 <- site10 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site11 <- site11 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site12 <- site12 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site13 <- site13 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site14 <- site14 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site15 <- site15 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site16 <- site16 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site17 <- site17 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site18 <- site18 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site19 <- site19 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site20 <- site20 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site21 <- site21 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site22 <- site22 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))
site23 <- site23 %>% filter((nb0 != 0 & nb1 != 0) | (nb1 != 0 & nb2 != 0))

```

# Calculate TAJIMA'S D
```{r}
n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site1$contig))) {
  df <- site1 %>% filter(contig == unique(site1$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site1 <- tibble(contig = unique(site1$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site1


n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site2$contig))) {
  df <- site2 %>% filter(contig == unique(site2$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site2 <- tibble(contig = unique(site2$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site2

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site3$contig))) {
  df <- site3 %>% filter(contig == unique(site3$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site3 <- tibble(contig = unique(site3$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site3

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site4$contig))) {
  df <- site4 %>% filter(contig == unique(site4$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site4 <- tibble(contig = unique(site4$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site4

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site5$contig))) {
  df <- site5 %>% filter(contig == unique(site5$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site5 <- tibble(contig = unique(site5$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site5

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site6$contig))) {
  df <- site6 %>% filter(contig == unique(site6$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site6 <- tibble(contig = unique(site6$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site6

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site7$contig))) {
  df <- site7 %>% filter(contig == unique(site7$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site7 <- tibble(contig = unique(site7$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site7

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site8$contig))) {
  df <- site8 %>% filter(contig == unique(site8$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site8 <- tibble(contig = unique(site8$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site8

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site9$contig))) {
  df <- site9 %>% filter(contig == unique(site9$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site9 <- tibble(contig = unique(site9$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site9

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site10$contig))) {
  df <- site10 %>% filter(contig == unique(site10$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site10 <- tibble(contig = unique(site10$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site10

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site11$contig))) {
  df <- site11 %>% filter(contig == unique(site11$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site11 <- tibble(contig = unique(site11$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site11

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site12$contig))) {
  df <- site12 %>% filter(contig == unique(site12$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site12 <- tibble(contig = unique(site12$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site12

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site13$contig))) {
  df <- site13 %>% filter(contig == unique(site13$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site13 <- tibble(contig = unique(site13$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site13


n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site14$contig))) {
  df <- site14 %>% filter(contig == unique(site14$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site14 <- tibble(contig = unique(site14$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site14

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site15$contig))) {
  df <- site15 %>% filter(contig == unique(site1$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site15 <- tibble(contig = unique(site15$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site15

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site16$contig))) {
  df <- site16 %>% filter(contig == unique(site16$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site16 <- tibble(contig = unique(site16$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site16

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site17$contig))) {
  df <- site17 %>% filter(contig == unique(site17$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site17 <- tibble(contig = unique(site17$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)

df_tajimaD_per_contigs_site17

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site18$contig))) {
  df <- site18 %>% filter(contig == unique(site1$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site18 <- tibble(contig = unique(site18$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site18

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site19$contig))) {
  df <- site19 %>% filter(contig == unique(site19$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site19 <- tibble(contig = unique(site19$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)

df_tajimaD_per_contigs_site19

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site20$contig))) {
  df <- site20 %>% filter(contig == unique(site20$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site20 <- tibble(contig = unique(site20$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site20
n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site21$contig))) {
  df <- site21 %>% filter(contig == unique(site21$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site21 <- tibble(contig = unique(site21$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site21

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site22$contig))) {
  df <- site22 %>% filter(contig == unique(site22$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site22 <- tibble(contig = unique(site22$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site22

n = 248
h = c(0) # heterozygosity
theta.pi = c(0)
theta.w = c(0)
tajimaD = c(0)
a1=sum(1/(seq(from=1, to=(n-1), by=1)))
variance.d <- function(n,S) {
    a1=sum(1/(seq(from=1, to=(n-1), by=1)))
    a2=sum(1/((seq(from=1, to=(n-1), by=1))**2))
    b1=(n+1)/(3*(n-1))
    b2=(2*((n**2)+n+3))/((9*n)*(n-1))
    c1=b1 - (1/a1)
    c2=b2-((n+2)/(a1*n)) + (a2/(a1**2))
    e1=c1/a1
    e2=c2/((a1**2)+a2)
    var=(e1*S) + (e2*S*(S-1))
    return(var)
}

for (i in 1:length(unique(site23$contig))) {
  df <- site23 %>% filter(contig == unique(site23$contig)[i])
  S <- nrow(df) # number of segregating sites of contigs
  h <- c(0)
  for (j in  1:nrow(df)) {
    h[j] <- 2*df$freq_ref[j]*df$freq_alt[j]*(n/(n-1)) # heterozygosities
  }   
  theta.pi[i] <- sum(h) # sum of heterozygosities all segregating sites per contig
  theta.w[i] <- S/a1
  tajimaD[i] <- (theta.pi[i] - theta.w[i]) / (sqrt(variance.d(n,S)))
}
df_tajimaD_per_contigs_site23 <- tibble(contig = unique(site23$contig),
                                     theta.w = theta.w,
                                     theta.pi = theta.pi,
                                     tajimaD = tajimaD)
df_tajimaD_per_contigs_site23

```

```{r}
df_tajimaD_per_contigs_site1 <- 
  df_tajimaD_per_contigs_site1 %>%
  mutate(site = "site1_319")

df_tajimaD_per_contigs_site2 <- 
  df_tajimaD_per_contigs_site2 %>%
  mutate(site = "site2_113")

df_tajimaD_per_contigs_site3 <- 
  df_tajimaD_per_contigs_site3 %>%
  mutate(site = "site3_111")

df_tajimaD_per_contigs_site4 <- 
  df_tajimaD_per_contigs_site4 %>%
  mutate(site = "site4_ColG")

df_tajimaD_per_contigs_site5 <- 
  df_tajimaD_per_contigs_site5 %>%
  mutate(site = "site5_335")

df_tajimaD_per_contigs_site6 <- 
  df_tajimaD_per_contigs_site6 %>%
  mutate(site = "site6_118")

df_tajimaD_per_contigs_site7 <- 
  df_tajimaD_per_contigs_site7 %>%
  mutate(site = "site7_168")

df_tajimaD_per_contigs_site8 <- 
  df_tajimaD_per_contigs_site8 %>%
  mutate(site = "site8_129")

df_tajimaD_per_contigs_site9 <- 
  df_tajimaD_per_contigs_site9 %>%
  mutate(site = "site9_163")

df_tajimaD_per_contigs_site10 <- 
  df_tajimaD_per_contigs_site10 %>%
  mutate(site = "site10_293")

df_tajimaD_per_contigs_site11 <- 
  df_tajimaD_per_contigs_site11 %>%
  mutate(site = "site11_223")

df_tajimaD_per_contigs_site12 <- 
  df_tajimaD_per_contigs_site12 %>%
  mutate(site = "site12_124")

df_tajimaD_per_contigs_site13 <- 
  df_tajimaD_per_contigs_site13 %>%
  mutate(site = "site13_324")

df_tajimaD_per_contigs_site14 <- 
  df_tajimaD_per_contigs_site14 %>%
  mutate(site = "site14_FRA")

df_tajimaD_per_contigs_site15 <- 
  df_tajimaD_per_contigs_site15 %>%
  mutate(site = "site15_CAB")

df_tajimaD_per_contigs_site16 <- 
  df_tajimaD_per_contigs_site16 %>%
  mutate(site = "site16_8")

df_tajimaD_per_contigs_site17 <- 
  df_tajimaD_per_contigs_site17 %>%
  mutate(site = "site17_Tour")

df_tajimaD_per_contigs_site18 <- 
  df_tajimaD_per_contigs_site18 %>%
  mutate(site = "site18_Pont")

df_tajimaD_per_contigs_site19 <- 
  df_tajimaD_per_contigs_site19 %>%
  mutate(site = "site19_pied_Hortus")

df_tajimaD_per_contigs_site20 <- 
  df_tajimaD_per_contigs_site20 %>%
  mutate(site = "site20_SA")

df_tajimaD_per_contigs_site21 <- 
  df_tajimaD_per_contigs_site21 %>%
  mutate(site = "site21_NewC")

# 70 km away
#df_tajimaD_per_contigs_site22 <- 
#  df_tajimaD_per_contigs_site22 %>%
#  mutate(site = "site22_70kmaway")

# t.embergeri
#df_tajimaD_per_contigs_site23 <- 
#  df_tajimaD_per_contigs_site23 %>%
#  mutate(site = "site23_embergeri")
```


```{r}
site_specific_tajimaD <- rbind(df_tajimaD_per_contigs_site1,
                               df_tajimaD_per_contigs_site2,
                               df_tajimaD_per_contigs_site3,
                               df_tajimaD_per_contigs_site4,
                               df_tajimaD_per_contigs_site5,
                               df_tajimaD_per_contigs_site6,
                               df_tajimaD_per_contigs_site7,
                               df_tajimaD_per_contigs_site8,
                               df_tajimaD_per_contigs_site9,
                               df_tajimaD_per_contigs_site10,
                               df_tajimaD_per_contigs_site11,
                               df_tajimaD_per_contigs_site12,
                               df_tajimaD_per_contigs_site13,
                               df_tajimaD_per_contigs_site14,
                               df_tajimaD_per_contigs_site15,
                               df_tajimaD_per_contigs_site16,
                               df_tajimaD_per_contigs_site17,
                               df_tajimaD_per_contigs_site18,
                               df_tajimaD_per_contigs_site19,
                               df_tajimaD_per_contigs_site20,
                               df_tajimaD_per_contigs_site21)
                               #df_tajimaD_per_contigs_site22,
                               #df_tajimaD_per_contigs_site23)
```

```{r}
site_specific_tajimaD %>% 
  group_by(site) %>% 
  ggplot(aes(tajimaD, fill = site)) +
  geom_histogram(bins = 70) + 
  xlim(-2, 5)

site_specific_tajimaD %>% 
  group_by(site) %>% 
  ggplot(aes(tajimaD)) + 
  geom_histogram(bins = 50, fill = "darkred", col = "black") + 
  xlim(-2, 5) + ylab("Frequency") +
  facet_wrap(~site, ncol =3) +
  ggtitle("Distribution of Tajima D per site")

```

```{r}
results_of_contigs_of_interest <- site_specific_tajimaD %>%  filter(contig == "FC:Contig71000" | contig == "FC:Contig12377" | contig == "FC:KR920616.1"  | contig == "FC:JX946357.1"  |contig == "FC:KM272331.1" | contig == "FC:KM272332.1"   | contig == "FC:JX946358.1" |contig == "AT:Contig4190" |contig == "AT:Contig660"|contig == "FC:KC461937.1" )
```

# Average TajimaD values of contigs of interest
```{r}
for (i in unique(site_specific_tajimaD$site)) {
  a = c(0)
  a = mean((results_of_contigs_of_interest %>% filter(site == i))$tajimaD)
  print(i)
  print(a)
}
```

# Distribuiton of TajimaD values per site
```{r}
results_of_contigs_of_interest %>% group_by(site) %>% ggplot(aes(tajimaD, fill = site)) + geom_histogram(bins = 30) + xlim(-2, 5)
results_of_contigs_of_interest %>% group_by(site) %>% ggplot(aes(tajimaD)) + geom_histogram(bins = 30, fill = "darkred", col = "black") + xlim(-2, 5) + facet_wrap(~site, ncol =3) +
    ggtitle("Distribution of Tajima D per site for Contig of Interests")

```

```{r}
for (i in unique(results_of_contigs_of_interest$contig)) {
  a = c(0)
  a = mean((results_of_contigs_of_interest %>% filter(contig == i))$tajimaD)
  print(i)
  print(a)
}
```
# TajimaD values of contigs of interest for each site
```{r}
for (i in unique(results_of_contigs_of_interest$site)) {
  
  print(results_of_contigs_of_interest %>% filter(site == i)) 
 
}
```

